File: app/src/main/java/it/paranoidsquirrels/idleguildmaster/storage/data/items/abstractClasses/Potion.java
package it.paranoidsquirrels.idleguildmaster.storage.data.items.abstractClasses;

import it.paranoidsquirrels.idleguildmaster.R;

/* JADX INFO: loaded from: classes3.dex */
public abstract class Potion extends Consumable {
    public static final int AGILITY = 10;
    public static final int CONSTITUTION = 0;
    public static final int DARKNESS = 8;
    public static final int DEFENSE = 4;
    public static final int DEXTERITY = 1;
    public static final int HEALTH = 3;
    public static final int IMMUNITY = 9;
    public static final int INTELLIGENCE = 2;
    public static final int MAGIC_DEFENSE = 5;
    public static final int PRECISION = 6;
    public static final int VICIOUSNESS = 7;

    public abstract int getPotionType();

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.items.abstractClasses.Consumable, it.paranoidsquirrels.idleguildmaster.storage.data.items.Item
    public int printType() {
        return R.string.type_potion;
    }

    @Override // it.paranoidsquirrels.idleguildmaster.storage.data.items.abstractClasses.Consumable
    public int printConsumeImage() {
        return R.drawable.consume_potion;
    }
}

File: app/src/main/kotlin/it/paranoidsquirrels/idleguildmaster/storage/data/items/abstractClasses/Potion.kt
package it.paranoidsquirrels.idleguildmaster.storage.data.items.abstractClasses

import it.paranoidsquirrels.idleguildmaster.R

abstract class Potion : Consumable() {
    companion object {
        const val AGILITY: Int = 10
        const val CONSTITUTION: Int = 0
        const val DARKNESS: Int = 8
        const val DEFENSE: Int = 4
        const val DEXTERITY: Int = 1
        const val HEALTH: Int = 3
        const val IMMUNITY: Int = 9
        const val INTELLIGENCE: Int = 2
        const val MAGIC_DEFENSE: Int = 5
        const val PRECISION: Int = 6
        const val VICIOUSNESS: Int = 7
    }

    abstract fun getPotionType(): Int

    override fun printType(): Int = R.string.type_potion

    override fun printConsumeImage(): Int = R.drawable.consume_potion
}
